babble-frontend/src/App.js

```diff
@@ -1,21 +1,297 @@
 import './App.css';
 import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
-import React from "react";
+import React, { useState, useCallback } from "react";
 import Home from "./pages/Home/Home.js";
 import Login from "./pages/Login/Login.js";
 import Post from "./pages/Post/Post.js";
 import Profile from "./pages/Profile/Profile.js";
+import axios from 'axios';
 
+// localStorage.setItem(
+//   "AuthToken",
+//   "Bearer FAKETOKEN#ASDOANSCOAOCBDIVB",
+// );
+let token = localStorage.getItem("AuthToken");
+const apiURL = "";
+
+let authAxios = axios.create({
+  baseURL: apiURL,
+  headers: {
+    authentication: token
+  }
+});
 
 function App() {
+  const [loginInfo, setLoginInfo] = useState({});
+
+  // authentication calls
+  const createUser = useCallback( async (body) => {
+    try {
+      const result = await axios.post(`/enter/signup`, body);
+      console.log(result);
+      alert("Account created! Please go to login.");
+      return true;
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  });
+
+  const loginUser = useCallback( async ({body, callback}) => {
+    try {
+      const result = await axios.post(`/enter/login`, body);
+      console.log(result);
+      localStorage.setItem(
+        "AuthToken",
+        JSON.stringify(result.data),
+      );
+      token = localStorage.getItem("AuthToken");
+      console.log(token);
+      authAxios = axios.create({
+        baseURL: apiURL,
+        headers: {
+          authentication: token
+        }
+      });
+      setLoginInfo(body);
+      localStorage.setItem(
+        "email",
+        body.email
+      );
+      localStorage.setItem(
+        "pswd",
+        body.password
+      );//not secure at all but im v lazy rn
+      callback();
+    } catch (err) {
+      console.log(err);
+      alert("Invalid Login")
```

Comment:
Suggestion: could do 'alert(err)' since that prints out specific error messages from backend about why login didn't work.